examples/bpi-monolith-test-suite/bpi/components/workgroup/workgroup.ts

```diff
@@ -6,21 +6,30 @@ export class Workgroup {
      name: string;
      participants: BpiSubject[] = [];
      worksteps: Workstep[] = [];
+     // Set used to store IDs and ensure each ID can only be used once
+     workstepIds: Set<string> = new Set("");
 
     constructor(name: string, id: string, owner: BpiSubject, worksteps: Workstep[],) {
-        this.name = name;
         this.id = id;
+        this.name = name;
         this.worksteps = worksteps;
         this.participants.push(owner);
     }
 
      addWorkstep(workstep: Workstep) {
-          this.worksteps.push(workstep);
+
+          //push workstep if ID is not yet in use
+          if(!this.workstepIds.has(workstep.id)) {
+               this.worksteps.push(workstep);
+          }
+
+          //add ID to running set
+          this.workstepIds.add(workstep.id);
```

Comment:
@Ybittan thanks for answer, I get that Set is used to make sure id is only used once. But in my opinion using Set here is overkill,  you can just ask if workstep exist, it will always be small enough set in this implementation - just my opinion, feel free to ignore :) 

But what happens if addWorkstep is called with existing id, this function will just ignore that? In that case, caller of this function would not be aware that his workstep was not added because he is using existing id. Some error should be returned.

I think it would be better to make sure ID is always unique when creating workstep, by using uuid for example, instead of checking it like this.